## Import library yang dibutuhkan

In [47]:
import pandas as pd


# EKSPLORASI DATA

## Membaca CSV file

In [48]:
data_path = 'C:/Users/Fikri/Documents/Data_Kpp'
df = pd.read_csv(data_path+'/daily_rent_detail.csv')

C:\Users\Fikri\AppData\Local\Temp\ipykernel_20992\2004395355.py:2: DtypeWarning: Columns (5,7) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv(data_path+'/daily_rent_detail.csv')


## List baris dan kolom yang berada di dalam dataset 

In [49]:
df

ride_id  rideable_type               started_at  \
0         946D42AD89539210    docked_bike      2020-05-30 17:25:29   
1         CC46FAAB662B8613    docked_bike      2020-05-09 14:42:04   
2         72F00B2FB833D6ED    docked_bike      2020-05-24 17:27:19   
3         4DFBE6AED989DF35    docked_bike      2020-05-27 15:29:52   
4         1AAFE6B4331AB9DF    docked_bike      2020-05-31 14:06:03   
...                    ...            ...                      ...   
16086667  117C27DBDD138B72  electric_bike  2024-08-01 08:10:22.293   
16086668  4774F4D630258482  electric_bike  2024-08-08 10:05:21.938   
16086669  D75836E25E77B5EC  electric_bike  2024-08-03 16:29:32.252   
16086670  3B888603D18116DC  electric_bike  2024-08-03 02:49:45.661   
16086671  3B201BA37CE6ED10  electric_bike  2024-08-15 07:51:42.628   

                         ended_at  \
0             2020-05-31 18:25:22   
1             2020-05-09 15:06:33   
2             2020-05-24 17:43:51   
3             2020-05-27 15:47:13   
4             2020-05-31 14:30:30   
...                           ...   
16086667  2024-08-01 08:10:37.641   
16086668  2024-08-08 10:21:46.654   
16086669  2024-08-03 16:35:43.179   
16086670  2024-08-03 02:59:56.877   
16086671  2024-08-15 07:55:50.645   

                                       start_station_name start_station_id  \
0                                       Anacostia Library            31804   
1                                          10th & E St NW            31256   
2         Connecticut Ave & Newark St NW / Cleveland Park            31305   
3         Connecticut Ave & Newark St NW / Cleveland Park            31305   
4                              Georgia Ave & Morton St NW            31419   
...                                                   ...              ...   
16086667                                              NaN              NaN   
16086668                                              NaN              NaN   
16086669                                              NaN              NaN   
16086670                                              NaN              NaN   
16086671                                              NaN              NaN   

                       end_station_name end_station_id  start_lat  start_lng  \
0                        11th & H St NE        31614.0  38.865784 -76.978400   
1         21st St & Constitution Ave NW        31261.0  38.895914 -77.026064   
2                        12th & U St NW        31268.0  38.934267 -77.057979   
3                  14th & Belmont St NW        31119.0  38.934267 -77.057979   
4                        17th & K St NW        31213.0  38.932128 -77.023500   
...                                 ...            ...        ...        ...   
16086667                            NaN            NaN  38.890000 -77.000000   
16086668                            NaN            NaN  38.900000 -77.000000   
16086669                            NaN            NaN  38.920000 -76.990000   
16086670                            NaN            NaN  38.920000 -77.020000   
16086671                            NaN            NaN  38.940000 -77.030000   

            end_lat    end_lng member_casual  
0         38.899983 -76.991383        casual  
1         38.892459 -77.046567        member  
2         38.916787 -77.028139        member  
3         38.921074 -77.031887        casual  
4         38.902760 -77.038630        casual  
...             ...        ...           ...  
16086667  38.890000 -77.000000        member  
16086668  38.870000 -76.950000        member  
16086669  38.920000 -77.000000        member  
16086670  38.920000 -77.030000        member  
16086671  38.930000 -77.030000        member  

[16086672 rows x 13 columns]

Ditunjukkan bahwa terdapat : 
- **16086672** baris
- **13** kolom

## Informasi general mengenai dataset

In [50]:
df.info()
df.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16086672 entries, 0 to 16086671
Data columns (total 13 columns):
 #   Column              Dtype  
---  ------              -----  
 0   ride_id             object 
 1   rideable_type       object 
 2   started_at          object 
 3   ended_at            object 
 4   start_station_name  object 
 5   start_station_id    object 
 6   end_station_name    object 
 7   end_station_id      object 
 8   start_lat           float64
 9   start_lng           float64
 10  end_lat             float64
 11  end_lng             float64
 12  member_casual       object 
dtypes: float64(4), object(9)
memory usage: 1.6+ GB


start_lat     start_lng       end_lat       end_lng
count  1.608666e+07  1.608666e+07  1.606074e+07  1.606074e+07
mean   3.890389e+01 -7.703200e+01  3.890290e+01 -7.703162e+01
std    2.758673e-02  3.436006e-02  4.704474e-02  8.250517e-02
min    3.873000e+01 -7.740000e+01  0.000000e+00 -7.810000e+01
25%    3.889054e+01 -7.704463e+01  3.889050e+01 -7.704468e+01
50%    3.890304e+01 -7.703150e+01  3.890257e+01 -7.703000e+01
75%    3.891554e+01 -7.701350e+01  3.891305e+01 -7.701237e+01
max    3.914000e+01 -7.678414e+01  4.307000e+01  0.000000e+00

**Bedasarkan observasi output diatas, didapatkan bahwa :**
terdapat 13 kolom yang berbeda, disertai dengan 2 tipe data yang berbeda dengan rincian sebagai berikut :
- 9 kolom (ride_id, rideable_type, started_at, ended_at, start_station_name, start_station_id, end_station_name, end_station_id, member_casual) dengan tipe data = **object**
- 4 kolom (start_lat, start_lng, end_lat, end_lng) dengan tipe data = **float64**

# DATA CLEANING

In [51]:
df.isnull().sum()

ride_id                     0
rideable_type               0
started_at                  0
ended_at                    0
start_station_name    1450150
start_station_id      1450150
end_station_name      1558268
end_station_id        1558917
start_lat                  10
start_lng                  10
end_lat                 25929
end_lng                 25929
member_casual               0
dtype: int64

setelah menjalankan *syntax* diatas, didapatkan bahwa terdapat kolom dengan nilai yang kosong. Untuk menghindari adanya kolom dengan nilai yang kosong pada *dataset*, akan dilakukan pengisian nilai '0' sebagai pengganti nilai kolom yang kosong dengan menggunakan *syntax* di bawah ini :

In [52]:
df.fillna(value=0, inplace=True)

In [53]:
# re-check
df.isnull().sum()

ride_id               0
rideable_type         0
started_at            0
ended_at              0
start_station_name    0
start_station_id      0
end_station_name      0
end_station_id        0
start_lat             0
start_lng             0
end_lat               0
end_lng               0
member_casual         0
dtype: int64

didapat bahwa sudah tidak ada nilai kosong pada *dataset*, yang berarti proses berhasil

## Menghapus Redundansi

In [57]:
df_new = df.duplicated
df_new.drop('start_station_name',axis=1,inplace=True)

AttributeError: 'function' object has no attribute 'drop'